services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core2/_context_base.py

```diff
@@ -43,10 +43,9 @@ class _ContextIOInterface(ABC):
     async def to_dict(self) -> dict[str, Any]:
         """returns the context of a store as a dictionary"""
 
-    @classmethod
     @abstractmethod
-    async def from_dict(cls, incoming: dict[str, Any]) -> "_ContextIOInterface":
-        """returns an instance from incoming deserialized data"""
+    async def from_dict(self, incoming: dict[str, Any]) -> None:
+        """stores data from incoming deserialized data"""
```

Comment:
I find strange that this is not a ``@classmethod``. Typically these ``from_`` are used as constructors .For instance

```python
a = A.from_dict(obj)
```

If you have an existing instance and you update its state you normally write:
```python
a = A()
a.update(obj)
```